### PR TITLE
updated documentation for tf.math.argmin and tf.math.argmax operations

### PR DESCRIPTION
**Pull request by mandroid6**
PR for issue #26530 and #26532

Added usage examples for tf.math.argmin and tf.math.argmax functions with this patch.
Updated docstring in tensorflow/python/ops/math_ops.py

### CODE REVIEW

**Review by mihaimaruseac — COMMENTED** [review]
Thanks for doing this, but can you make all updates consistent across argmin and argmax? Same words, same spacing, same format.

**Comment by mandroid6** [comment]
Yes, will make the change today.

**Comment by mandroid6** [comment]
@mihaimaruseac have made the changes

File: tensorflow/core/ops/math_ops.cc [hunk 1]
```diff
@@ -969,7 +969,11 @@ REGISTER_OP("ArgMax")
     .Attr("T: numbertype")
     .Attr("Tidx: {int32, int64} = DT_INT32")
     .Attr("output_type: {int32, int64} = DT_INT64")
-    .SetShapeFn(ArgOpShape);
+    .SetShapeFn(ArgOpShape)
+	.Doc(R"doc(Returns the index with the largest value across dimensions of a tensor.
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: You have `dimensions` here and `axes` below.

File: tensorflow/core/ops/math_ops.cc [hunk 2]
```diff
@@ -978,7 +982,10 @@ REGISTER_OP("ArgMin")
     .Attr("T: numbertype")
     .Attr("Tidx: {int32, int64} = DT_INT32")
     .Attr("output_type: {int32, int64} = DT_INT64")
-    .SetShapeFn(ArgOpShape);
+    .SetShapeFn(ArgOpShape)
+	.Doc(R"doc(Returns the index with the smallest value across axes of a tensor.
+Note that in case of ties the identity of the return value is not guaranteed
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Missing `.` at the end

File: tensorflow/core/ops/math_ops.cc [hunk 3]
```diff
@@ -978,7 +982,10 @@ REGISTER_OP("ArgMin")
     .Attr("T: numbertype")
     .Attr("Tidx: {int32, int64} = DT_INT32")
     .Attr("output_type: {int32, int64} = DT_INT64")
-    .SetShapeFn(ArgOpShape);
+    .SetShapeFn(ArgOpShape)
+	.Doc(R"doc(Returns the index with the smallest value across axes of a tensor.
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Should also add the `inputs: Must all..` line to match the`ArgMax` docstring

File: tensorflow/core/api_def/base_api/api_def_ArgMin.pbtxt [hunk 4]
```diff
@@ -11,5 +11,8 @@ END
   summary: "Returns the index with the smallest value across dimensions of a tensor."
   description: <<END
 Note that in case of ties the identity of the return value is not guaranteed.
+
+For example:
+
```
Comment (mihaimaruseac) [inline tensorflow/core/api_def/base_api/api_def_ArgMin.pbtxt]: `ArgMax.pbtxt` has a full example listed under `Usage`, this one doesn't and also seems incomplete.

File: tensorflow/core/api_def/base_api/api_def_ArgMax.pbtxt [hunk 5]
```diff
@@ -11,5 +11,17 @@ END
   summary: "Returns the index with the largest value across dimensions of a tensor."
   description: <<END
 Note that in case of ties the identity of the return value is not guaranteed.
+
+Usage:
+
+	```python
```
Comment (mihaimaruseac) [inline tensorflow/core/api_def/base_api/api_def_ArgMax.pbtxt]: I think you have extra spacing here compared to the python docstring (the code block here starts a few columns after `:` whereas in the Python files it starts on the same column as `:`)

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — DISMISSED** [review]
Thank you

**Comment by mandroid6** [comment]
@mihaimaruseac for the thorough comments! Will take care going forward :)

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by mihaimaruseac** [comment]
This will need changes as internal tests fail with

```Invalid argument: No matching input/output/attr for name 'inputs' from Doc() for Op ArgMax```

File: tensorflow/core/ops/math_ops.cc [hunk 6]
```diff
@@ -969,7 +969,11 @@ REGISTER_OP("ArgMax")
     .Attr("T: numbertype")
     .Attr("Tidx: {int32, int64} = DT_INT32")
     .Attr("output_type: {int32, int64} = DT_INT64")
-    .SetShapeFn(ArgOpShape);
+    .SetShapeFn(ArgOpShape)
+	.Doc(R"doc(Returns the index with the largest value across axes of a tensor.
+inputs: Must all be the same size and shape.
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Let's remove the `inputs: Must all..` lines, both here and below in `ArgMin`. These are no longer arguments to the op and having them here causes the builds to crash.

Comment (mandroid6) [reply tensorflow/core/ops/math_ops.cc]: Done

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
There is another failure (`bazel test //tensorflow/core/api_def:api_test`)

```
Value of: op.summary().empty()
  Actual: false
Expected: true
OpDef for ArgMax has a doc string. Doc strings must be defined in ApiDef instead of OpDef. Please, add summary and descriptions in api_def_ArgMax.pbtxt file instead
[  FAILED  ] BaseApiTest.OpDefsShouldNotHaveDocs (74 ms)
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
See new failure, doc strings shouldn't go in OpDefs

**Comment by mandroid6** [comment]
@mihaimaruseac have removed the doctring from OpDefs

File: tensorflow/core/ops/math_ops.cc [hunk 7]
```diff
@@ -968,17 +968,17 @@ REGISTER_OP("ArgMax")
...
 REGISTER_OP("ArgMin")
     .Input("input: T")
     .Input("dimension: Tidx")
     .Output("output: output_type")
     .Attr("T: numbertype")
     .Attr("Tidx: {int32, int64} = DT_INT32")
-    .Attr("output_type: {int32, int64} = DT_INT64")
-    .SetShapeFn(ArgOpShape);
+    .Attr("output_type: {int32, int64} = DT_INT64");
+
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Shouldn't remove the shape function

File: tensorflow/core/ops/math_ops.cc [hunk 8]
```diff
@@ -968,17 +968,17 @@ REGISTER_OP("ArgMax")
...
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Shouldn't remove the shape function

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by mandroid6** [comment]
Done, had mistakenly deleted the shape function

File: tensorflow/core/ops/math_ops.cc [hunk 9]
```diff
@@ -970,6 +970,7 @@ REGISTER_OP("ArgMax")
     .Attr("Tidx: {int32, int64} = DT_INT32")
     .Attr("output_type: {int32, int64} = DT_INT64")
     .SetShapeFn(ArgOpShape);
+
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Remove this extra empty line, styling tools will fail otherwise.

File: tensorflow/core/ops/math_ops.cc [hunk 10]
```diff
@@ -980,6 +981,7 @@ REGISTER_OP("ArgMin")
     .Attr("output_type: {int32, int64} = DT_INT64")
     .SetShapeFn(ArgOpShape);
 
+
```
Comment (mihaimaruseac) [inline tensorflow/core/ops/math_ops.cc]: Remove this extra empty line, styling tools will fail otherwise.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/math_ops.py [hunk 11]
```diff
@@ -205,6 +215,16 @@ def argmin_v2(input,
 
   Returns:
     A `Tensor` of type `output_type`.
+	
+  Usage:
+    ```python
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/math_ops.py]: This line should start on the same column as `:` above

Comment (mandroid6) [reply tensorflow/python/ops/math_ops.py]: I don't know what's happening. I had made the same indentation for both. Now made the changes

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**